pytorch_lightning/loops/closure.py

```diff
@@ -0,0 +1,113 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Callable, Optional
+
+from torch import Tensor
+
+from pytorch_lightning.profiler import BaseProfiler
+from pytorch_lightning.trainer.connectors.logger_connector.result import ResultCollection
+from pytorch_lightning.utilities.warnings import WarningCache
+
+
+@dataclass
+class ClosureResult:
+    closure_loss: Optional[Tensor]
+    result_collection: Optional[ResultCollection]
+
+    @property
+    def loss(self) -> Optional[Tensor]:
+        if self.closure_loss is not None:
+            return self.closure_loss.detach().clone()
```

Comment:
Clone in `make_backward_fn` and here?

And what if you don't want to detach?

Reply:
Yes I know. This is just here ​to stick to how it was previously handled. I don't yet know how to handle this cleanly without breaking stuff.